src/JwtMethod.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Auth\Jwt;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Yiisoft\Auth\AuthenticationMethodInterface;
+use Yiisoft\Auth\IdentityInterface;
+use Yiisoft\Auth\IdentityRepositoryInterface;
+use Yiisoft\Http\Header;
+
+/**
+ * Authentication method based on JWT token.
+ */
+final class JwtMethod implements AuthenticationMethodInterface
+{
+    private string $headerName = Header::AUTHORIZATION;
+    private string $queryParamName = 'access-token';
+    private string $headerTokenPattern = '/^Bearer\s+(.*?)$/';
+    private string $realm = 'api';
+    private string $identifier = 'sub';
+    private IdentityRepositoryInterface $identityRepository;
+    private TokenManagerInterface $tokenManager;
+
+    public function __construct(IdentityRepositoryInterface $identityRepository, TokenManagerInterface $tokenManager)
+    {
+        $this->identityRepository = $identityRepository;
+        $this->tokenManager = $tokenManager;
+    }
+
+    public function authenticate(ServerRequestInterface $request): ?IdentityInterface
+    {
+        $token = $this->getAuthenticationToken($request);
+        $claims = $this->tokenManager->getClaims($token);
+
+        if (isset($claims[$this->identifier])) {
```

Comment:
Here somewhere there should be a check that the token is not expired?

Reply:
Yes.